derp/derp_test.go

```diff
@@ -475,7 +475,7 @@ func newTestServer(t *testing.T) *testServer {
 			// TODO: register c in ts so Close also closes it?
 			go func(i int) {
 				brwServer := bufio.NewReadWriter(bufio.NewReader(c), bufio.NewWriter(c))
-				go s.Accept(c, brwServer, fmt.Sprintf("test-client-%d", i))
+				go s.Accept(context.Background(), c, brwServer, fmt.Sprintf("test-client-%d", i))
```

Comment:
this should be something scoped to the test. So maybe do something like at the start of the test and then use `ctx` here.
```
ctx, cancel := context.WithCancel(context.Background())
defer cancel()
```